test_conformance/run_conformance.py

```diff
@@ -63,19 +66,33 @@ def get_tests(filename, devices_to_test):
         comment = re.search("^#.*", line)
         if comment:
             continue
-        device_specific_match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*,\s*(.+?)\s*$", line)
+        if sys.version_info[0] < 3:
+            device_specific_match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*,\s*(.+?)\s*$", line)
+        else:
+            device_specific_match = re.search("^\s*(.+)\s*,\s*(.+)\s*,\s*(.+)\s*", line)
         if device_specific_match:
             if device_specific_match.group(1) in devices_to_test:
-                test_path = string.replace(device_specific_match.group(3), '/', os.sep)
-                test_name = string.replace(device_specific_match.group(2), '/', os.sep)
+                if sys.version_info[0] < 3:
+                    test_path = string.replace(device_specific_match.group(3), '/', os.sep)
+                    test_name = string.replace(device_specific_match.group(2), '/', os.sep)
+                else:
+                    test_path = str.replace(device_specific_match.group(3), '/', os.sep)
+                    test_name = str.replace(device_specific_match.group(2), '/', os.sep)
                 tests.append((test_name, test_path))
             else:
                 print("Skipping " + device_specific_match.group(2) + " because " + device_specific_match.group(1) + " is not in the list of devices to test.")
             continue
-        match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*$", line)
+        if sys.version_info[0] < 3:
+            match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*$", line)
```

Comment:
As above, I'm confused by this regex change.  Could you explain it please?

Reply:
See above.

Reply:
Regex changes removed.